XPlat/AzureMessaging/build.cake

```diff
@@ -1,48 +1,127 @@
+var IOS_VERSION = "2.0.4";
+var IOS_NUGET_VERSION = IOS_VERSION;
+var IOS_URL = $"https://github.com/Azure/azure-notificationhubs-ios/releases/download/{IOS_VERSION}/WindowsAzureMessaging.framework.zip";
 
-var IOS_VERSION = "423aaba626e3eccdc4770bee8861a8ab8518563b";
-var IOS_NUGET_VERSION = "1.2.5.2";
-var IOS_URL = string.Format ("https://github.com/Azure/azure-notificationhubs/raw/{0}/iOS/bin/WindowsAzureMessaging.framework.zip", IOS_VERSION);
-
-var ANDROID_VERSION = "0.5";
-var ANDROID_NUGET_VERSION = "0.5.0";
+var ANDROID_VERSION = "0.6";
+var ANDROID_NUGET_VERSION = "0.6.0";
 var ANDROID_URL = string.Format ("https://dl.bintray.com/microsoftazuremobile/SDK/com/microsoft/azure/notification-hubs-android-sdk/{0}/notification-hubs-android-sdk-{0}.aar", ANDROID_VERSION);
 
-var buildSpec = new BuildSpec {
-
-	Libs = new [] {
-		new DefaultSolutionBuilder {
-			SolutionPath = "./iOS/source/Xamarin.Azure.NotificationHubs.iOS.sln",
-			OutputFiles = new [] { 
-				new OutputFileCopy { FromFile = "./iOS/source/bin/unified/Release/Xamarin.Azure.NotificationHubs.iOS.dll" },
-			}
-		},
-		new DefaultSolutionBuilder {
-			SolutionPath = "./Android/source/Xamarin.Azure.NotificationHubs.Android.sln",
-			OutputFiles = new [] { 
-				new OutputFileCopy { FromFile = "./Android/source/bin/Release/Xamarin.Azure.NotificationHubs.Android.dll" },
-			}
-		},
-	},
-
-	Samples = new [] {
-		new IOSSolutionBuilder { SolutionPath = "./iOS/samples/NotificationHubsSampleiOS.sln",  Configuration = "Release", Platform="iPhone" },
-		new DefaultSolutionBuilder { SolutionPath = "./Android/samples/NotificationHubsSampleAndroid.sln" },
-	},
-
-	NuGets = new [] {
-		new NuGetInfo { NuSpec = "./nuget/Xamarin.Azure.NotificationHubs.iOS.nuspec", Version = IOS_NUGET_VERSION, RequireLicenseAcceptance = true  },
-		new NuGetInfo { NuSpec = "./nuget/Xamarin.Azure.NotificationHubs.Android.nuspec", Version = ANDROID_NUGET_VERSION , RequireLicenseAcceptance = true },
-	},
-
-	Components = new [] {
-		new Component { ManifestDirectory = "./component" },
-	}
-};
+Task("libs-ios")
+	.WithCriteria(IsRunningOnUnix())
+	.IsDependentOn("externals-ios")
+	.Does (() =>
+{
+	MSBuild("./iOS/source/Xamarin.Azure.NotificationHubs.iOS.sln", c => {
+		c.Configuration = "Release";
+		c.Restore = true;
+		c.Targets.Clear();
+		c.Targets.Add("Rebuild");
+		c.Properties.Add("DesignTimeBuild", new [] { "false" });
+		c.BinaryLogger = new MSBuildBinaryLogSettings {
+			Enabled = true,
+			FileName = "./output/libs.binlog"
+		};
+	});
+});
+
+Task("libs-android")
+	.IsDependentOn("externals-android")
+	.Does (() =>
+{
+	MSBuild("./Android/source/Xamarin.Azure.NotificationHubs.Android.sln", c => {
+		c.Configuration = "Release";
+		c.Restore = true;
+		c.Targets.Clear();
+		c.Targets.Add("Rebuild");
+		c.Properties.Add("DesignTimeBuild", new [] { "false" });
+		c.BinaryLogger = new MSBuildBinaryLogSettings {
+			Enabled = true,
+			FileName = "./output/libs.binlog"
```

Comment:
Maybe rename this to `libs-android` and `libs-ios` for iOS. Same with the samples and nuget variants